server/services/userService.ts

```diff
@@ -1,19 +1,24 @@
 import logger from '../../logger'
-import type { CaseloadClient, HmppsAuthClient, RestClientBuilder, RestClientBuilderWithoutToken } from '../data'
+import type { CaseloadClient, HmppsManageUsersClient, RestClientBuilder } from '../data'
 import { StringUtils } from '../utils'
 import type { UserDetails } from '@accredited-programmes/users'
 import type { Caseload } from '@prison-api'
 
 export default class UserService {
   constructor(
-    private readonly hmppsAuthClientBuilder: RestClientBuilderWithoutToken<HmppsAuthClient>,
+    private readonly hmppsManageUsersClientBuilder: RestClientBuilder<HmppsManageUsersClient>,
     private readonly caseloadClientBuilder: RestClientBuilder<CaseloadClient>,
   ) {}
 
-  async getUser(token: Express.User['token']): Promise<UserDetails> {
-    const hmppsAuthClient = this.hmppsAuthClientBuilder()
-    const user = await hmppsAuthClient.getUser(token)
-    const caseloads = await this.getCaseloads(token)
+  async getUserWithDetails(token: Express.User['token']): Promise<UserDetails> {
+    const hmppsManageUsersClient = this.hmppsManageUsersClientBuilder(token)
+    const { username } = await hmppsManageUsersClient.getCurrentUsername()
+
+    const [user, caseloads] = await Promise.all([
+      hmppsManageUsersClient.getUserFromUsername(username),
+      this.getCaseloads(token),
+    ])
```

Comment:
Is this format preferable to just having two `await` lines (like before)?

Reply:
Yeah I hadn't done it anywhere else up to now (although not sure if there's been a reason to) but I tend to prefer it as it runs them concurrently.

Reply:
Ooh nice! Might be good to start doing that then